scandium-core/src/main/java/org/eclipse/californium/scandium/dtls/Handshaker.java

```diff
@@ -292,8 +297,8 @@ protected Handshaker(boolean isClient, int initialMessageSeq, DTLSSession sessio
 		this.pskStore = config.getPskStore();
 		this.session.setMaxTransmissionUnit(maxTransmissionUnit);
 		this.applicationLevelInfoSupplier = config.getApplicationLevelInfoSupplier();
-		this.deferredApplicationData = new ArrayList<RawData>(maxDeferredProcessedApplicationDataMessages);
-		this.deferredRecords = new ArrayList<Record>(maxDeferredProcessedApplicationDataMessages);
+		this.deferredApplicationData = new ArrayList<RawData>(maxDeferredProcessedOutgoingApplicationDataMessages);
+		this.deferredRecords = new ArrayList<Record>(maxDeferredProcessedOutgoingApplicationDataMessages);
```

Comment:
Should we initialize this arraylist with the max size ?
If I'm not wrong, with default value this makes 64KB (8192*2*4 bytes) used for each Handshaker.
My guess is that most of the time the order of magniture is most arround 10.

Reply:
Hm, my intention was for outgoing the number of messages, default 10.
For incoming my intention was the amount of bytes. But the RecordsSize is not used to initialize the arrays. So I have no idea, why it's 64K (or 819224 )?

Reply:
My understanding for incoming stuff:
Max 8K for deferred records.
Max 8k for fragmented records
Sum => 16K per handshake

Reply:
I read this too quickly, I believe that default value for `maxDeferredProcessedOutgoingApplicationDataMessages` was `8192` ... :confounded: 
(I lost myself with all those `maxDeferred` variable :sweat_smile: )

Default value is 10, so initializing arrayList to `maxDeferredProcessedOutgoingApplicationDataMessages` is OK with default value.
This will cost 80B(10*2*4bytes) by handshaker by default.

But this makes me think that if somebody choose a very large value, he could consume lot of RAM because we initialize arrayList to this size instead of just let start with default capacity (using default constructor `new ArrayList()`)

Reply:
OK